Update the documentation (#15)
fastlane/github-actions: CONTRIBUTING.md

## Development

> First of all, thank you for your interested in this project. Please read the documentation here to being contributing. 

Reviewer: ```suggestion
First of all, thank you for your interest in this project. Please read the documentation here to begin contributing. 
```

### Creating an Action

- :file_folder: In a new branch create an new folder in a root folder of this repository and name it to your Action name.
- :octocat: Follow the instructions [Creating an Action with the Toolkit](https://github.com/actions/toolkit#creating-an-action-with-the-toolkit) to create an Action.
- :construction: Start developing and open a pull request.

Read more about GitHub Actions [here](https://help.github.com/en/articles/getting-started-with-github-actions).

### Recommendations on versioning your action

Since we keep multiple actions in one repository, all the actions are released in one batch (we do not support semantic versioning). 

**JavaScript / TypeScript Action**

> When your pull request is merged:

Reviewer: After or while it is being merged?
Are these instructions to prepare a release?
Why does a PR contributor have to do that? Isn't that part of a release process?

Author: > After or while it is being merged?

When your pull request was merged. 

> Are these instructions to prepare a release?

Basically yes 👍  

> Why does a PR contributor have to do that? Isn't that part of a release process?

Should, the project maintainers could do that. Where should we put a release description then? 🤔

Reviewer: Section "How to release new versions" and then maybe a "Steps to prepare the code" and "How to do the actual release" or similar 👍

Author: I changed the section title to `How to release new versions`. I think splitting the others will be a bit confusing. Right now we have two path to release the action based on its type: node Action or Docker Action.  WDYT?

- create a new branch
- comment out `node_modules` in `.gitignore`

```
# comment out in distribution branches
# node_modules/
```

> - add **only** your production dependencies

Reviewer: ```suggestion
- add **only** your production dependencies:
```

```bash
$ npm prune --production
$ git add node_modules
$ git commit -a -m "Add production dependencies"
$ git push
```

> - open a pull request with a destionation branch to `latest` branch

Reviewer: What does that mean exactly?

```suggestion
- open a pull request with a destination branch to `latest` branch
```

## Development

> First of all, thank you for your interest in this project. Please read the documentation here to begin contributing. 

Reviewer 2: ```suggestion
Thank you for your interest in this project. Please read the documentation here to begin contributing. 
```

### Creating an Action

- :file_folder: In a new branch create an new folder in a root folder of this repository and name it to your Action name.
- :octocat: Follow the instructions [Creating an Action with the Toolkit](https://github.com/actions/toolkit#creating-an-action-with-the-toolkit) to create an Action.
- :construction: Start developing and open a pull request.

Read more about GitHub Actions [here](https://help.github.com/en/articles/getting-started-with-github-actions).

### Recommendations on versioning your action

> Since we keep multiple actions in one repository, all the actions are released in one batch (we do not support semantic versioning yet). 

Reviewer 2: Should these “actions” be capitalized here?

**JavaScript / TypeScript Action**

> When your pull request was merged:

Reviewer: This should say "To prepare the code for a new release:" or similar.

> When your pull request was merged:

Reviewer 2: Does this need to say something like “To start your pull request:”?

- create a new branch from `master` branch
- comment out `node_modules` in `.gitignore`

```
# comment out in distribution branches
# node_modules/

> ```

Reviewer: By adding three spaces in front of the code block (all lines)  you might be able to put this code block _in_ the list

```suggestion
   ```
```
(but also for all other lines - which GH does not support yet afaik)

- add **only** your production dependencies:

```bash
$ npm prune --production
$ git add node_modules
$ git commit -a -m "Add production dependencies"
$ git push
```

- open a pull request with a destination branch to `latest` branch

> - everything is ready to be :ship:

Reviewer: ```suggestion
- everything is ready to be shipped :ship:
```

**Docker Action**

> When your pull request was merged:

Reviewer 2: Same here. Does this need to say “To start your pull request:”?

Author: > When your pull request was merged:

Sorry this one is not clear :// The idea is that when a pull request with an Action implementation is merged into `master` branch, we are ready to release it. To release this Action, we need to open a pull request to the `latest` branch. @janpio pointed out that maybe it should be out of scope for the pull request author. I agree with that 👍 

So you can think of it as instructions for releasing new Action.  Maybe we should create another doc file like `RELEASE.md` or something and put `Recommendations on versioning your action` section there, thought? 🤔

- create a new branch from `master` branch
- open a pull request with a destination branch to `latest` branch

> - everything is ready to be :ship:

Reviewer: ```suggestion
- everything is ready to be shipped :ship:
```
